rstudio/rstudio
# Require rmarkdown for rmd and qmd if any R code chunks are found

### Intent

Addresses [pro#3310](https://github.com/rstudio/rstudio-pro/issues/3310) -- only scan quarto docs for R package dependencies if any R code chunks are found, and if so, always add rmarkdown to the list of required packages so users can get the package install prompt. Follow up to #10767.

### Approach

We have an extractRcode function, but it was a little broken. We use this to extract any R code chunks from a document (specifically Rmd and qmd, for .R files we extract the entire file) and since we're already checking the file contents for mentions of any packages we need to be installed, we add rmarkdown to the list -- but only if the R content is non empty.
That way users get the package install prompt if they have at least 1 R chunk, but if they have no code chunks, or only python chunks, we won't check their R package dependency situation.

### Automated Tests

None

### QA Notes

`remove.packages("rmarkdown")` for testing
Check a quarto doc with at least 1 R code chunk, and check a quarto doc with no R code chunks (may have python chunks for example)

Python only
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/7817881/160695152-74dea00a-1be9-4b7a-b4c6-a8d6f4a9fc74.png">

R only
<img width="1122" alt="image" src="https://user-images.githubusercontent.com/7817881/160695185-8a335175-ed2a-4ba8-879e-08ccbedc1d1e.png">

### Checklist
